tests/Unit/API/Site/Controllers/Ads/CampaignControllerTest.php

```diff
@@ -147,14 +156,55 @@ public function test_create_campaign_invalid_country_code() {
 		$response = $this->do_request( self::ROUTE_CAMPAIGNS, 'POST', $campaign_data );
 
 		$this->assertEquals( 'rest_invalid_param', $response->get_data()['code'] );
-		$this->assertEquals( 'Invalid parameter(s): country', $response->get_data()['message'] );
+		$this->assertEquals( 'Invalid parameter(s): targeted_locations', $response->get_data()['message'] );
+		$this->assertEquals( 400, $response->get_status() );
+	}
+
+	public function test_create_campaign_unsupported_country() {
+		$this->country_supported = false;
+		$this->google_helper->method( 'is_country_supported' )
+			->willReturnCallback( function () { return $this->country_supported; } );
+
+		$campaign_data = [
+			'amount'             => 20,
+			'targeted_locations' => ['CN', 'KP'],
+		];
+
+		$response = $this->do_request( self::ROUTE_CAMPAIGNS, 'POST', $campaign_data );
+
+		$this->assertEquals( 'rest_invalid_param', $response->get_data()['code'] );
+		$this->assertEquals( 'Invalid parameter(s): targeted_locations', $response->get_data()['message'] );
 		$this->assertEquals( 400, $response->get_status() );
 	}
 
+	public function test_create_campaign_empty_targeted_locations() {
```

Comment:
From my understanding of the design we should always be passing at least one targeted country when we create a new campaign, but maybe I'm seeing that wrong. Which means this test should fail with a validation error.

Normally we could use `'minItems' => 1` in the schema, but since we use a custom validation function that would have to be evaluated within the validation function. The default validation gets the schema from the request, [see](https://developer.wordpress.org/reference/functions/rest_validate_request_arg/). But it might make sense to always require at least one country if it's an array, so we might not need to access the schema.

Reply:
I think it makes sense, added empty check in the country code validation function via c9ea09e. Also tested on other controllers that use the same validator and seems fine. But should keep an eye on them if it causes problem.